zio-http/src/main/scala/zio/http/Response.scala

```diff
@@ -14,13 +14,16 @@ import zio.{Cause, Promise, Task, Trace, Unsafe, ZIO}
 
 import java.io.IOException
 import zio.stacktracer.TracingImplicits.disableAutoTrace // scalafix:ok;
+import java.util.concurrent.atomic.AtomicReference
 
-final case class Response private (
+sealed abstract case class Response private (
   status: Status,
   headers: Headers,
   body: Body,
   private[zio] val attribute: Response.Attribute,
   private[zio] val httpError: Option[HttpError],
+  private[zio] val encodedResponse: AtomicReference[Option[Response.EncodedResponse]],
+  frozen: Boolean = false,
```

Comment:
I think you should delete all the private variables, and make it a `trait` that emulates a `case class`, including `copy`, `hashCode`, `==`, `apply`, and `unapply`. 

Then add a `def cached: Response` that returns a subtype like `Response.Cached`. The cached response can store `Map[Any, Any]` and drivers can, if a response is cached, choose to accelerate rendering by precomputing the response.

Alternately, there is no need for `Map[Any, Any]` if the Netty driver uses its own map, which can be a map of `Response` to `Response.EncodedResponse`.

Reply:
I already implemented the second alternative with a `ConcurrentMap`.